src/sinks/gcp/cloud_storage.rs

```diff
@@ -0,0 +1,543 @@
+use super::{healthcheck_response, GcpAuthConfig, GcpCredentials, Scope};
+use crate::{
+    event::{self, Event},
+    sinks::{
+        util::{
+            http::{https_client, HttpsClient},
+            retries::{RetryAction, RetryLogic},
+            tls::{TlsOptions, TlsSettings},
+            BatchBytesConfig, Buffer, PartitionBuffer, PartitionInnerBuffer, ServiceBuilderExt,
+            SinkExt, TowerRequestConfig,
+        },
+        Healthcheck, RouterSink,
+    },
+    template::Template,
+    topology::config::{DataType, SinkConfig, SinkContext, SinkDescription},
+};
+use bytes::Bytes;
+use chrono::Utc;
+use futures::{stream::iter_ok, Future, Poll, Sink};
+use http::{Method, StatusCode, Uri};
+use hyper::{
+    header::{HeaderName, HeaderValue},
+    Body, Request,
+};
+use lazy_static::lazy_static;
+use serde::{Deserialize, Serialize};
+use snafu::Snafu;
+use std::collections::HashMap;
+use tower::{Service, ServiceBuilder};
+use tracing::field;
+use uuid::Uuid;
+
+const NAME: &str = "gcp_cloud_storage";
+const BASE_URL: &str = "https://storage.googleapis.com/";
+
+#[derive(Clone)]
+struct GcsSink {
+    bucket: String,
+    client: HttpsClient,
+    creds: Option<GcpCredentials>,
+    base_url: String,
+    settings: RequestSettings,
+}
+
+#[derive(Debug, Snafu)]
+enum GcsError {
+    #[snafu(display("Bucket {:?} not found", bucket))]
+    BucketNotFound { bucket: String },
+}
+
+#[derive(Deserialize, Serialize, Debug, Default)]
+#[serde(deny_unknown_fields)]
+pub struct GcsSinkConfig {
+    bucket: String,
+    acl: Option<GcsPredefinedAcl>,
+    storage_class: Option<GcsStorageClass>,
+    tags: Option<HashMap<String, String>>,
+    pub key_prefix: Option<String>,
```

Comment:
Why are some of these pub and some are not?

Reply:
Historical accident. Most existing configs use `pub` for all of their fields, and I had originally just cloned this source from `aws_s3`. However, anything I had to add or modify, I dropped the `pub`, as it appears to be unnecessary. I have just dropped all of the `pub` markers in this sink, and it builds and runs.